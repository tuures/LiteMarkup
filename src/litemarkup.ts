import { parseToAst } from './parseToAst'
import { astToHtml } from './astToHtml'

//
// Facade
//

export * from './ast'
export * from './parseToAst'
export * from './astToHtml'

export const mdToHtml = (src: string) => astToHtml(parseToAst(src))

