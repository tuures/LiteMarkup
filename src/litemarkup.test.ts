import { parseToAst, astToHtml } from './litemarkup'

test('basic', () => {
  const src = `
### fooðŸ˜€

gadasdasd,asdasd
> quote here

***

asdas<b>d</b>asda sda1&amp;

ffofðŸ˜€oofoof
fofoofo2
# immediate heading

This is a list:
- one
- two
- three
- fourth item

  I mean this is a big one

- fifth

1. first
2. second
3. third

werwerwerwer

  \`\`\`\`clara
  foo = bar
   bar = zot
   1 < 2
  \`\`\`\`

immediate code:
\`\`\`
  foo = bar
\`\`\`

\`\`\`
\`\`\`

\`\`\`justnewline">eviltext

\`\`\`

> # asd
> lol <nottag>
lazy
> foo

> level 1
> > level 2
> > level 2

<div>
 foobar
</div>

<broken>
oops </broken> html
</broken>

joo\` asd \` bar

asd \`\`0doo

asd \` \`1doo

asd \`  \`2doo

asd \`   \`3doo

asd \`foo\`\` 12dii

asd \`\`foo\` 21dii

asd \`\`foo\`\` 22dii

asd\`asd\\
xfoobar

asd\\\`asd\\
yfoobar

br in the end \\

*I'm bold*

_I'm italic_

*bold outside and _italic inside_*

_bold start

bold ending_

aftert this comes a few paragraphs with just one character

a

#

_

aftert this comes a text content that is just one characters

a*bold*

_italic_s

\\!\\"\\#\\$\\%\\&\\'\\(\\)\\*\\+\\,\\-\\.\\/\\:\\;\\<\\=\\>\\?\\@\\[\\\\\\]\\^\\_\\\`\\{\\|\\}\\~

\\â†’\\A\\a\\ \\3\\Ï†\\Â«

\\*not emphasized*

*emphasized star\\**

_\\_italic underlines\\__

_italic underline\\_and star*_

\\\\*emphasis*

*notbold\`*\`

this paragraph has a [link to](/thisurl) and some parenthesis []() that is not a link

[[foo\\]a](ds\\)fd\\)\\\`)

![image of a cat](/ordidnthappen.jpg)

`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('markdown compatibility mode', () => {
  const src = `
**I'm bold**
__I'm bold too__

_I'm italic_
*I'm italic too*

these are supported:
**bold outside and _italic inside_**
__bold outside and *italic inside*__
_italic outside and **bold inside**_
*italic outside and __bold inside__*

these are not supported:
**bold outside and *italic inside***
__bold outside and _italic inside___
_italic outside and __bold inside___
*italic outside and **bold inside***

_not italic

not italic_

a**bold**

_italic_s

\\*not emphasized*

*emphasized star\\**

_\\_italic underlines\\__

_italic underline\\_and star*_

\\\\*emphasis after a literal backslash*

*just plain p before code\`*\`

`
  const ast = parseToAst(src, true)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  console.log(html)
  expect(html).toMatchSnapshot()
});

test('multiple images in series', () => {
  const src = `
![image of a cat](/ordidnthappen.jpg) and
second ![image of a cat](/ordidnthappen.jpg)
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('list with marker character in content', () => {
  const src = `
- foo - the first item
- bar-second
- baz- the third
- bam -fourth
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('nested lists', () => {
  const src = `
- foo
  - foo1
  - foo2
- bar
   * bar1
   * bar2
* another list
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codeblock without end', () => {
  const src = `
\`\`\`foo
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codeblock with broken end', () => {
  const src = `
\`\`\`foo
\`\`
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codeblock with broken end on the same line / codespan with missing backticks in the end', () => {
  const src = `
\`\`\`foo\`\`
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codespan with extra backticks on the start', () => {
  const src = `
foo \`\`\`foo\`\`
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codespan with extra backticks on the end', () => {
  const src = `
\`\`foo\`\`\`
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('codespan with three backticks in the beginning of a paragraph (not codeblock)', () => {
  const src = `
\`\`\`foo\`\`\`
`
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});

test('empty paragraph in the end of the file', () => {
  const src = 'foo\n  '
  const ast = parseToAst(src)
  expect(ast).toMatchSnapshot()

  const html = astToHtml(ast)
  expect(html).toMatchSnapshot()
});
