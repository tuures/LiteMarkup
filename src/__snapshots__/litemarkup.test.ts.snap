// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Array [
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "fooðŸ˜€",
      },
    ],
    "level": 3,
    "name": "h",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "gadasdasd,asdasd",
      },
    ],
    "name": "p",
  },
  Object {
    "doc": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "quote here",
          },
        ],
        "name": "p",
      },
    ],
    "name": "bq",
  },
  Object {
    "name": "hr",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asdas<b>d</b>asda sda1&amp;",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "ffofðŸ˜€oofoof
fofoofo2",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "immediate heading",
      },
    ],
    "level": 1,
    "name": "h",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "This is a list:",
      },
    ],
    "name": "p",
  },
  Object {
    "items": Array [
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "one",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "two",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "three",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "fourth item",
              },
            ],
            "name": "p",
          },
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "I mean this is a big one",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "fifth",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
    ],
    "name": "l",
    "startNumber": undefined,
  },
  Object {
    "items": Array [
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "first",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "second",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "third",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
    ],
    "name": "l",
    "startNumber": 1,
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "werwerwerwer",
      },
    ],
    "name": "p",
  },
  Object {
    "infoText": "clara",
    "name": "cb",
    "txt": "foo = bar
 bar = zot
 1 < 2
",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "immediate code:",
      },
    ],
    "name": "p",
  },
  Object {
    "infoText": "",
    "name": "cb",
    "txt": "  foo = bar
",
  },
  Object {
    "infoText": "",
    "name": "cb",
    "txt": "",
  },
  Object {
    "infoText": "justnewline\\">eviltext",
    "name": "cb",
    "txt": "
",
  },
  Object {
    "doc": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "asd",
          },
        ],
        "level": 1,
        "name": "h",
      },
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "lol <nottag>
lazy
foo",
          },
        ],
        "name": "p",
      },
    ],
    "name": "bq",
  },
  Object {
    "doc": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "level 1",
          },
        ],
        "name": "p",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "level 2
level 2",
              },
            ],
            "name": "p",
          },
        ],
        "name": "bq",
      },
    ],
    "name": "bq",
  },
  Object {
    "name": "htm",
    "raw": "<div>
 foobar
</div>",
  },
  Object {
    "name": "htm",
    "raw": "<broken>
oops </broken> html
</broken>",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "joo",
      },
      Object {
        "name": "cs",
        "txt": "asd",
      },
      Object {
        "name": "",
        "txt": " bar",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "",
        "txt": "\`\`0doo",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "cs",
        "txt": " ",
      },
      Object {
        "name": "",
        "txt": "1doo",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "cs",
        "txt": "  ",
      },
      Object {
        "name": "",
        "txt": "2doo",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "cs",
        "txt": "   ",
      },
      Object {
        "name": "",
        "txt": "3doo",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "cs",
        "txt": "foo",
      },
      Object {
        "name": "",
        "txt": "\` 12dii",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "",
        "txt": "\`\`foo",
      },
      Object {
        "name": "",
        "txt": "\` 21dii",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd ",
      },
      Object {
        "name": "cs",
        "txt": "foo",
      },
      Object {
        "name": "",
        "txt": " 22dii",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd",
      },
      Object {
        "name": "",
        "txt": "\`asd",
      },
      Object {
        "name": "br",
      },
      Object {
        "name": "",
        "txt": "xfoobar",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "asd",
      },
      Object {
        "name": "",
        "txt": "\`",
      },
      Object {
        "name": "",
        "txt": "asd",
      },
      Object {
        "name": "br",
      },
      Object {
        "name": "",
        "txt": "yfoobar",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "br in the end ",
      },
      Object {
        "name": "br",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "I'm bold",
          },
        ],
        "name": "b",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "I'm italic",
          },
        ],
        "name": "i",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "bold outside and ",
          },
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "italic inside",
              },
            ],
            "name": "i",
          },
        ],
        "name": "b",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "_bold start",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "bold ending",
      },
      Object {
        "name": "",
        "txt": "_",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "aftert this comes a few paragraphs with just one character",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "a",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "#",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "_",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "aftert this comes a text content that is just one characters",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "a",
      },
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "bold",
          },
        ],
        "name": "b",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "italic",
          },
        ],
        "name": "i",
      },
      Object {
        "name": "",
        "txt": "s",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "!",
      },
      Object {
        "name": "",
        "txt": "\\"",
      },
      Object {
        "name": "",
        "txt": "#",
      },
      Object {
        "name": "",
        "txt": "$",
      },
      Object {
        "name": "",
        "txt": "%",
      },
      Object {
        "name": "",
        "txt": "&",
      },
      Object {
        "name": "",
        "txt": "'",
      },
      Object {
        "name": "",
        "txt": "(",
      },
      Object {
        "name": "",
        "txt": ")",
      },
      Object {
        "name": "",
        "txt": "*",
      },
      Object {
        "name": "",
        "txt": "+",
      },
      Object {
        "name": "",
        "txt": ",",
      },
      Object {
        "name": "",
        "txt": "-",
      },
      Object {
        "name": "",
        "txt": ".",
      },
      Object {
        "name": "",
        "txt": "/",
      },
      Object {
        "name": "",
        "txt": ":",
      },
      Object {
        "name": "",
        "txt": ";",
      },
      Object {
        "name": "",
        "txt": "<",
      },
      Object {
        "name": "",
        "txt": "=",
      },
      Object {
        "name": "",
        "txt": ">",
      },
      Object {
        "name": "",
        "txt": "?",
      },
      Object {
        "name": "",
        "txt": "@",
      },
      Object {
        "name": "",
        "txt": "[",
      },
      Object {
        "name": "",
        "txt": "\\\\",
      },
      Object {
        "name": "",
        "txt": "]",
      },
      Object {
        "name": "",
        "txt": "^",
      },
      Object {
        "name": "",
        "txt": "_",
      },
      Object {
        "name": "",
        "txt": "\`",
      },
      Object {
        "name": "",
        "txt": "{",
      },
      Object {
        "name": "",
        "txt": "|",
      },
      Object {
        "name": "",
        "txt": "}",
      },
      Object {
        "name": "",
        "txt": "~",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "\\\\â†’",
      },
      Object {
        "name": "",
        "txt": "\\\\A",
      },
      Object {
        "name": "",
        "txt": "\\\\a",
      },
      Object {
        "name": "",
        "txt": "\\\\ ",
      },
      Object {
        "name": "",
        "txt": "\\\\3",
      },
      Object {
        "name": "",
        "txt": "\\\\Ï†",
      },
      Object {
        "name": "",
        "txt": "\\\\Â«",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "*",
      },
      Object {
        "name": "",
        "txt": "not emphasized",
      },
      Object {
        "name": "",
        "txt": "*",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "\\\\",
      },
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "emphasis",
          },
        ],
        "name": "b",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "*notbold",
      },
      Object {
        "name": "cs",
        "txt": "*",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "name": "",
        "txt": "this paragraph has a ",
      },
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "link to",
          },
        ],
        "href": "/thisurl",
        "name": "a",
      },
      Object {
        "name": "",
        "txt": " and some parenthesis ",
      },
      Object {
        "name": "",
        "txt": "[]() that is not a link",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "body": Array [
          Object {
            "name": "",
            "txt": "[foo",
          },
          Object {
            "name": "",
            "txt": "]",
          },
          Object {
            "name": "",
            "txt": "a",
          },
        ],
        "href": "ds\\\\)fd\\\\)\\\\\`",
        "name": "a",
      },
    ],
    "name": "p",
  },
  Object {
    "body": Array [
      Object {
        "alt": "image of a cat",
        "name": "img",
        "src": "/ordidnthappen.jpg",
      },
    ],
    "name": "p",
  },
]
`;

exports[`basic 2`] = `
"<h3>fooðŸ˜€</h3>
<p>gadasdasd,asdasd</p>
<blockquote>
  <p>quote here</p>
</blockquote>
<hr/>
<p>asdas&lt;b&gt;d</b>asda sda1&amp;amp;</p>
<p>ffofðŸ˜€oofoof
fofoofo2</p>
<h1>immediate heading</h1>
<p>This is a list:</p>
<ul>
  <li>
    <p>one</p>
  </li>
  <li>
    <p>two</p>
  </li>
  <li>
    <p>three</p>
  </li>
  <li>
    <p>fourth item</p>
    <p>I mean this is a big one</p>
  </li>
  <li>
    <p>fifth</p>
  </li>
</ul>
<ol start=\\"1\\">
  <li>
    <p>first</p>
  </li>
  <li>
    <p>second</p>
  </li>
  <li>
    <p>third</p>
  </li>
</ol>
<p>werwerwerwer</p>
<pre><code data-infotext=\\"clara\\">foo = bar
 bar = zot
 1 &lt; 2
</code></pre>
<p>immediate code:</p>
<pre><code>  foo = bar
</code></pre>
<pre><code/></pre>
<pre><code data-infotext=\\"justnewline&quot;&gt;eviltext\\">
</code></pre>
<blockquote>
  <h1>asd</h1>
  <p>lol &lt;nottag&gt;
lazy
foo</p>
</blockquote>
<blockquote>
  <p>level 1</p>
  <blockquote>
    <p>level 2
level 2</p>
</blockquote>
</blockquote>
<div>
 foobar
</div>
<broken>
oops </broken> html
</broken>
<p>joo<code>asd</code> bar</p>
<p>asd \`\`0doo</p>
<p>asd <code> </code>1doo</p>
<p>asd <code>  </code>2doo</p>
<p>asd <code>   </code>3doo</p>
<p>asd <code>foo</code>\` 12dii</p>
<p>asd \`\`foo\` 21dii</p>
<p>asd <code>foo</code> 22dii</p>
<p>asd\`asd<br/>xfoobar</p>
<p>asd\`asd<br/>yfoobar</p>
<p>br in the end <br/></p>
<p><span style=\\"font-weight: bold;\\">I&apos;m bold</span></p>
<p><span style=\\"font-style: italic;\\">I&apos;m italic</span></p>
<p><span style=\\"font-weight: bold;\\">bold outside and <span style=\\"font-style: italic;\\">italic inside</span></span></p>
<p>_bold start</p>
<p>bold ending_</p>
<p>aftert this comes a few paragraphs with just one character</p>
<p>a</p>
<p>#</p>
<p>_</p>
<p>aftert this comes a text content that is just one characters</p>
<p>a<span style=\\"font-weight: bold;\\">bold</span></p>
<p><span style=\\"font-style: italic;\\">italic</span>s</p>
<p>!&quot;#$%&amp;&apos;()*+,-./:;&lt;=&gt;?@[\\\\]^_\`{|}~</p>
<p>\\\\â†’\\\\A\\\\a\\\\ \\\\3\\\\Ï†\\\\Â«</p>
<p>*not emphasized*</p>
<p>\\\\<span style=\\"font-weight: bold;\\">emphasis</span></p>
<p>*notbold<code>*</code></p>
<p>this paragraph has a <a href=\\"/thisurl\\">link to</a> and some parenthesis []() that is not a link</p>
<p><a href=\\"ds\\\\)fd\\\\)\\\\\`\\">[foo]a</a></p>
<p><img alt=\\"image of a cat\\" src=\\"/ordidnthappen.jpg\\"/></p>"
`;

exports[`list with marker character in content 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "foo - the first item",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "bar-second",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "baz- the third",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "bam -fourth",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
    ],
    "name": "l",
    "startNumber": undefined,
  },
]
`;

exports[`list with marker character in content 2`] = `
"<ul>
  <li>
    <p>foo - the first item</p>
  </li>
  <li>
    <p>bar-second</p>
  </li>
  <li>
    <p>baz- the third</p>
  </li>
  <li>
    <p>bam -fourth</p>
  </li>
</ul>"
`;

exports[`multiple images in series 1`] = `
Array [
  Object {
    "body": Array [
      Object {
        "alt": "image of a cat",
        "name": "img",
        "src": "/ordidnthappen.jpg",
      },
      Object {
        "name": "",
        "txt": " and
second ",
      },
      Object {
        "alt": "image of a cat",
        "name": "img",
        "src": "/ordidnthappen.jpg",
      },
    ],
    "name": "p",
  },
]
`;

exports[`multiple images in series 2`] = `
"<p><img alt=\\"image of a cat\\" src=\\"/ordidnthappen.jpg\\"/> and
second <img alt=\\"image of a cat\\" src=\\"/ordidnthappen.jpg\\"/></p>"
`;

exports[`nested lists 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "foo",
              },
            ],
            "name": "p",
          },
          Object {
            "items": Array [
              Object {
                "doc": Array [
                  Object {
                    "body": Array [
                      Object {
                        "name": "",
                        "txt": "foo1",
                      },
                    ],
                    "name": "p",
                  },
                ],
                "name": "li",
              },
              Object {
                "doc": Array [
                  Object {
                    "body": Array [
                      Object {
                        "name": "",
                        "txt": "foo2",
                      },
                    ],
                    "name": "p",
                  },
                ],
                "name": "li",
              },
            ],
            "name": "l",
            "startNumber": undefined,
          },
        ],
        "name": "li",
      },
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "bar",
              },
            ],
            "name": "p",
          },
          Object {
            "items": Array [
              Object {
                "doc": Array [
                  Object {
                    "body": Array [
                      Object {
                        "name": "",
                        "txt": "bar1",
                      },
                    ],
                    "name": "p",
                  },
                ],
                "name": "li",
              },
              Object {
                "doc": Array [
                  Object {
                    "body": Array [
                      Object {
                        "name": "",
                        "txt": "bar2",
                      },
                    ],
                    "name": "p",
                  },
                ],
                "name": "li",
              },
            ],
            "name": "l",
            "startNumber": undefined,
          },
        ],
        "name": "li",
      },
    ],
    "name": "l",
    "startNumber": undefined,
  },
  Object {
    "items": Array [
      Object {
        "doc": Array [
          Object {
            "body": Array [
              Object {
                "name": "",
                "txt": "another list",
              },
            ],
            "name": "p",
          },
        ],
        "name": "li",
      },
    ],
    "name": "l",
    "startNumber": undefined,
  },
]
`;

exports[`nested lists 2`] = `
"<ul>
  <li>
    <p>foo</p>
    <ul>
      <li>
        <p>foo1</p>
      </li>
      <li>
        <p>foo2</p>
      </li>
    </ul>
  </li>
  <li>
    <p>bar</p>
    <ul>
      <li>
        <p>bar1</p>
      </li>
      <li>
        <p>bar2</p>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>
    <p>another list</p>
  </li>
</ul>"
`;
